python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py

```diff
@@ -35,42 +36,74 @@ def _get_node_name(node_info: Mapping[str, Any]):
 
 
-def _dbt_node_to_asset(
-    node_info: Mapping[str, Any],
+def _dbt_nodes_to_assets(
+    node_infos: Sequence[Mapping[str, Any]],
     runtime_metadata_fn: Optional[
         Callable[[SolidExecutionContext, Mapping[str, Any]], Mapping[str, Any]]
     ] = None,
     io_manager_key: Optional[str] = None,
-) -> OpDefinition:
-    code_block = textwrap.indent(node_info["raw_sql"], "    ")
-    description_sections = [
-        node_info["description"],
-        "#### Columns:\n" + _columns_to_markdown(node_info["columns"])
-        if len(node_info["columns"]) > 0
-        else None,
-        f"#### Raw SQL:\n```\n{code_block}\n```",
-    ]
-    description = "\n\n".join(filter(None, description_sections))
-
-    @asset(
-        name=node_info["name"],
-        description=description,
-        non_argument_deps={
-            AssetKey(dep_name.split(".")[-1]) for dep_name in node_info["depends_on"]["nodes"]
-        },
+) -> AssetsDefinition:
+    outs = {}
+    sources = set()
+    out_name_to_node_info = {}
+    for node_info in node_infos:
+        node_name = node_info["name"]
+        in_deps = []
+        out_deps = []
+        for dep in node_info["depends_on"]["nodes"]:
+            dep_type = dep.split(".")[0]
+            dep_name = dep.split(".")[-1]
+            if dep_type == "source":
+                sources.add(AssetKey(dep_name))
+                in_deps.append(dep_name)
+            elif dep_type == "model":
+                out_deps.append(dep_name)
+        code_block = textwrap.indent(node_info["raw_sql"], "    ")
+        description_sections = [
+            node_info["description"],
+            "#### Columns:\n" + _columns_to_markdown(node_info["columns"])
+            if len(node_info["columns"]) > 0
+            else None,
+            f"#### Raw SQL:\n```\n{code_block}\n```",
+        ]
+        description = "\n\n".join(filter(None, description_sections))
+
+        outs[node_name] = Out(
+            dagster_type=None,
+            asset_key=AssetKey(node_name),
+            in_deps=in_deps,
+            out_deps=out_deps,
+            description=description,
+            io_manager_key=io_manager_key,
+        )
+        out_name_to_node_info[node_name] = node_info
+
+    @multi_asset(
+        name="dbt_project",
+        non_argument_deps=sources,
+        outs=outs,
         required_resource_keys={"dbt"},
-        io_manager_key=io_manager_key,
         compute_kind="dbt",
     )
-    def _node_asset(context):
-        context.resources.dbt.run(models=[".".join([node_info["package_name"], node_info["name"]])])
-        if runtime_metadata_fn:
-            metadata = runtime_metadata_fn(context, node_info)
-            yield Output(None, metadata=metadata)
-        else:
-            yield Output(None)
-
-    return _node_asset
+    def _dbt_project_multi_assset(context):
+        dbt_output = context.resources.dbt.run()
```

Comment:
Would this break the case where for example there is a python asset interleaved with DBT assets since everything is executed in one run? 

(imagine i have a source in dbt that reads a table generated by a python asset, which in turn depends on a dbt model, as in `dbt -> python -> dbt`)